src/pyk/kastManip.py

```
@@ -465,8 +484,14 @@ def minimize_term(term: KInner, keep_vars: Iterable[str] = (), abstract_labels:
     term = inline_cell_maps(term)
     term = remove_semantic_casts(term)
     term = useless_vars_to_dots(term, keep_vars=keep_vars)
-    term = labels_to_dots(term, abstract_labels)
+
+    if keep_cells:
+        term = extract_cells(term, keep_cells)
+    else:
+        term = labels_to_dots(term, abstract_labels)
+
     term = collapse_dots(term)
```

> If an empty term is returned (for example with --keep-cells "nonexistent_cell", then this is not handled and an exception is thrown in other part of the code. Question: should I catch this or no?

My guess is that this is the part that raises the exception - if so, `collapse_dots` should be fixed so that `collapse_dots(DOTS) == DOTS `. After all, the term `...` is minimal, so it can be returned as value.

---

Actually, that is not the case. `collapse_dots(DOTS) == DOTS` holds. The thing is that `minimize_term` is used in `__main__.py` where we actually assume that the result contains a config cell because we apply the result to the `split_config_and_constraints` function. This does not have to occur, as ML terms do not care specifically about cells. 

To keep the universality of `minimize_term` for any ML terms, I think the correct solution is to add a `try .. catch` block in this place. If it can be applied, then the printing algorithm works correctly. If it can't, then we can return an exception (that would also handle the case where we are left with something like `... #And PATH_COND) or simply return identity in these cases.

---

I'd prefer to handle this case explicitly with `if` rather than exception handling. Can this be easily done?

---

I understand. However, I can only think of doing the same thing as the check that is done in `split_config_and_constraints`, which would lead to duplicate code.

Or we can edit `split_config_and_constraints` to do something else than throw an exception. For example, return a tuple with `[EMPTY_K, constraints]` in a case like this and check for that.

---

Exception handling is fine then.